src/Extension/PublishableExtension.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of the Silverback API Component Bundle Project
+ *
+ * (c) Daniel West <daniel@silverback.is>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Silverback\ApiComponentBundle\Extension;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryCollectionExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\Common\Annotations\Reader;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use Silverback\ApiComponentBundle\Annotation\Publishable;
+use Symfony\Component\ExpressionLanguage\Expression;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+/**
+ * @author Vincent Chalamon <vincent@les-tilleuls.coop>
+ */
+final class PublishableExtension implements QueryItemExtensionInterface, QueryCollectionExtensionInterface
+{
+    private AuthorizationCheckerInterface $authorizationChecker;
+    private Reader $reader;
+    private ManagerRegistry $registry;
+    private RequestStack $requestStack;
+    private string $permission;
+    private ?Publishable $configuration;
+
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker, Reader $reader, ManagerRegistry $registry, RequestStack $requestStack, string $permission)
+    {
+        $this->authorizationChecker = $authorizationChecker;
+        $this->reader = $reader;
+        $this->registry = $registry;
+        $this->requestStack = $requestStack;
+        $this->permission = $permission;
+    }
+
+    public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, array $identifiers, string $operationName = null, array $context = []): void
+    {
+        $configuration = $this->getConfiguration($resourceClass);
+        if (!$configuration || !($request = $this->requestStack->getCurrentRequest()) || $request->isMethod(Request::METHOD_DELETE)) {
```

Comment:
I think this may give a user without access to drafts the ability to delete a draft as we are not applying the restrictions to the query